icu4c/source/i18n/collationdatabuilder.cpp

```diff
@@ -389,7 +389,6 @@ CollationDataBuilder::setPrimaryRangeAndReturnNext(UChar32 start, UChar32 end,
             primary = Collation::incThreeBytePrimaryByOffset(primary, isCompressible, step);
             if(start > end) { return primary; }
         }
-        modified = TRUE;
     }
```

Comment:
@daniel-ju Removing this seems a bit odd, as the method is named "set*" and it does look like this method can mutate the object.
What was the warning for this line?

Reply:
The warning was that the line was unreachable because of returning in the infinite for loop a few lines before this. Should we instead move this line before the loop in the else condition?

Reply:
Hmm, good question. I'm not entirely sure. It feels like perhaps the intent was for the modified to *always* be set if the code goes down the else path -- but I'm not sure.

I think it might be better to omit this change from the warning clean-up effort, and then file a separate ticket/issue just for this specific case. (As I'm not sure what the implications of the change would be).

Reply:
I'll revert this change and file a separate ticket for it.